### ISSUE TITLE
[Feature request] Ability to specify which .env file to use

### ISSUE THREAD

**Issue by johnwilhite** [JeffreyWay/laravel-mix#1446]
- Laravel Mix Version: 2.0.9
- Node Version: v8.4.0
- NPM Version: 5.5.1
- OS: macOS 10.13.2 

### Description:

When running `npm run production`, I'd like to be able to use a different file from `.env` (`.env.production`) to specify the MIX_ environment variables.  Is this possible currently?  I see the following in the source, but the only time the `envPath` param is set is in the tests.

https://github.com/JeffreyWay/laravel-mix/blob/master/src/plugins/MixDefinitionsPlugin.js#L10
```js
function MixDefinitionsPlugin(envPath) {
    expand(
        dotenv.config({
            path: envPath || Mix.paths.root('.env')
        })
    );
}
```

I'm currently swapping the env files with a bash script when building for production, but it seems like this could be made cleaner fairly easily with a mix option.

Thanks!

**Comment by bruno-fernandes** [comment]
@johnwilhite try this:

`require('dotenv').config({
    path: Mix.paths.root('.env-test')
});`

**Comment by johnwilhite** [comment]
@bruno-fernandes Thanks for the suggestion, but if I include that before declaring mix, it is overwritten by the above configuration.  If I include it afterwards it doesn't seem to affect anything during the build since the mix object is already built.

**Comment by johnwilhite** [comment]
@bruno-fernandes I started looking into this a little more and the reason your suggestion isn't working is dotenv will not overwrite env variables by default.  By the time the config file is loaded, all the environment variables have already been set and dotenv will not overwrite them, but suggests doing this: https://www.npmjs.com/package/dotenv#what-happens-to-environment-variables-that-were-already-set.

I ended up creating a super simple mix plugin to accomplish this.  https://github.com/johnwilhite/mix-env-file
